[sen5x] Extend with SEN6X support esphome

mikelawrence opened the pull request:
## Description:

Add support SEN6X sensors.

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** 

- esphome/esphome#9254

## Checklist:

  - [ ] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [x] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [ ] Link added in `/components/index.rst` when creating new documents for new components or cookbook.

mikelawrence committed:
Add SEN6X support to sen5x component

mikelawrence committed:
Minor changes

mikelawrence committed:
Document model configuration variable

Add model (now required)
Fix some default value errors
Add variable types

mikelawrence committed:
Update CO2 compensation

mikelawrence committed:
Fix formatting errors

mikelawrence committed:
Fix error in configuration (automatic_self_calibration)

Reorganize configuration examples
Add to note about warm-up time for the SEN6X models

mikelawrence committed:
Remove defaults to true from store_baseline; more cleanup

mikelawrence committed:
grammar and improved readability for the heater section

mikelawrence committed:
Revert "grammar and improved readability for the heater section"

This reverts commit 344833f558fa8a5cbcb1a3dd5708371556febc15.

mikelawrence committed:
Revert "Remove defaults to true from store_baseline; more cleanup"

This reverts commit d2aac49919be6fe180d189c73f4be4426439be7a.

mikelawrence committed:
Revert "Fix error in configuration (automatic_self_calibration)"

This reverts commit 982126d79c81beed8be01bcaa773dfb26f40bdc6.

mikelawrence committed:
Revert "Fix formatting errors"

This reverts commit 5ec3c62e74c396ca902a6a7b81adf48c5d138dfb.

mikelawrence committed:
Revert "Update CO2 compensation"

This reverts commit db6650dcb3c314d83fb70031957c7269038279fe.

mikelawrence committed:
Revert "Document model configuration variable"

This reverts commit 7be0f151f4375ca64ea2a124126b612514f4c4dc.

mikelawrence committed:
Revert "Minor changes"

This reverts commit 799b1c41283dc649b6c258ea2fa9abf38905b8f4.

mikelawrence committed:
Revert "Add SEN6X support to sen5x component"

This reverts commit 3479fffe7f868074520f2202d748298b42a73e7e.

mikelawrence committed:
Merge branch 'next' into current

mikelawrence committed:
Add sen6x documentation sen5x component

mikelawrence committed:
Lint corrections and more formatting

mikelawrence committed:
Remove double top headings; update images

mikelawrence committed:
Resize images

mikelawrence committed:
Resize images again

mikelawrence commented:
I have updated the document to use the new Markdown structure and the changes from PR #5176.

mikelawrence committed:
Spelling correction

danielkent-net commented on content/components/sensor/sen5x.md:
```diff
@@ -186,23 +235,140 @@ to blow out the accumulated dust inside the fan.
-The default automatic-cleaning interval is 168 hours (1 week) of uninterrupted use. Switching off the sensor resets
-this time counter.
+SEN6X sensors supports fan cleaning but not the automatic fan cleaning interval. You have to trigger
+`sen5x.start_fan_autoclean` action below occasionally.
```
Per https://github.com/esphome/esphome/pull/9254#issuecomment-3304145174, I think it would be a good idea to provide an example of how to trigger SEN6x autocleaning periodically in esphome. Suggestion:

```suggestion
`sen5x.start_fan_autoclean` action below occasionally. One way this can be implemented within ESPHome is by adding an `interval` component to perform this action periodically. For example, to clean the fan every 7 days while the device is on, as recommended by the manufacturer, the following configuration can be added:
    
    interval:
        - interval: 7d
          then:
            - sen5x.start_fan_autoclean: sen66_01
```

mikelawrence replied:
This is a good idea! I've added it.

mikelawrence committed:
Update sen5x.md

Add section about emulating SEN5X auto fan cleaning for the SEN6X sensors

mikelawrence committed:
Linting fixes

mikelawrence committed:
Merge branch 'next' into current

mikelawrence committed:
Merge branch 'next' into current

mikelawrence committed:
Update links and minor editing

mikelawrence committed:
Update note of start-up times for sensors

mikelawrence committed:
Refactor sensor reporting notes in sen5x.md

Reformatted sensor reporting notes for clarity.

mikelawrence committed:
Linting fix

mikelawrence committed:
Merge branch 'next' into current

mikelawrence committed:
Merge branch 'next' into current

mikelawrence committed:
Lots of changes to support recent datasheet updates

mikelawrence committed:
Minor cleanup

mikelawrence committed:
Updates actions and examples. Update SEN62 and SEN69C model changes

mikelawrence committed:
Merge branch 'next' into current

mikelawrence committed:
Fix formatting error

mikelawrence committed:
Fix links. And the inevitable edits that happen every time this document is opened.

mikelawrence committed:
Fix one more anchor link

mikelawrence committed:
Hopefully adding two columns at the top

mikelawrence committed:
Trying Hugo column shortcode instead

mikelawrence committed:
Another column approach

mikelawrence committed:
One more attempt at columns plus a test image

mikelawrence committed:
Fix dangling shortcode

mikelawrence committed:
Change heading levels

mikelawrence committed:
Go back to Hugo image link

mikelawrence committed:
Add more documentation on actions

mikelawrence committed:
Add link to temperature compensation section and more edits

mikelawrence committed:
Merge branch 'next' into current

mikelawrence committed:
General improvements
Add CO2 Compensation link in CO2 sensor section
Add notes about NOx unused algorithm parameters
Move Automatic Fan Cleaning section

mikelawrence committed:
Merge branch 'next' into current

danielkent-net commented on content/components/sensor/sen5x.md:
````diff
@@ -1,217 +1,478 @@
+    co2:
+      name: "CO₂"
+```
+
````
I think it would also be a good idea to also provide an example using a [user-defined action](https://esphome.io/components/api/#user-defined-actions). Something like this:

```yaml
api:
  encryption:
    key: !secret encryption_key
  actions:
    - action: sen69c_recalibrate_co2
      variables:
        co2: int
      then:
        - if:
            condition:
              lambda: 'return (co2 >= 400) and (co2 <= 1200);'
            then:
              - sen5x.perform_forced_co2_recalibration:
                  value: !lambda 'return co2;'
                  id: sen69c
              - api.respond:
                  success: true
            else:
              - api.respond:
                  success: false
                  error_message: "CO₂ input calibration must be between 400 and 1200."
```

You can test this added action by going into the developer console and searching for the action by name.

mikelawrence replied:
This is a great idea but I'm away for a few days. I will get to this by the end of the week.

mikelawrence commented:
Devs have decided to go with another PR
